Tests/AppTests/AppTests.swift

```diff
@@ -104,6 +104,34 @@ final class AppTests: XCTestCase {
         XCTAssertEqualJSON(update, expected, "Update Request JSON Mismatch")
     }
 
+    func testJiraErrors() throws {
+        let decoder = JSONDecoder()
+
+        let errorResponse1 = #"""
+            {"errorMessages":["Issue does not exist or you do not have permission to see it.","Some other message"],"errors":{}}
+        """#.data(using: .utf8)!
+
+        let error1 = try decoder.decode(JiraService.ServiceError.self, from: errorResponse1)
+        XCTAssertEqual(error1.errorMessages, ["Issue does not exist or you do not have permission to see it.", "Some other message"])
+        XCTAssertEqual(error1.errors, [:])
+        XCTAssertEqual(error1.reason, "Issue does not exist or you do not have permission to see it. ; Some other message")
+
+        let errorResponse2 = #"""
+            {"errorMessages":[],"errors":{"fixVersions":"Field 'fixVersions' cannot be set. It is not on the appropriate screen, or unknown."}}
+        """#.data(using: .utf8)!
+
+        let error2 = try decoder.decode(JiraService.ServiceError.self, from: errorResponse2)
+        XCTAssertEqual(error2.errorMessages, [])
+        XCTAssertEqual(error2.errors, ["fixVersions": "Field 'fixVersions' cannot be set. It is not on the appropriate screen, or unknown."])
+        XCTAssertEqual(error2.reason, "fixVersions: Field 'fixVersions' cannot be set. It is not on the appropriate screen, or unknown.")
+
+        let errorResponse3 = #"""
+            {"errorMessages":["msg1","msg2"],"errors":{"key1":"error1","key2":"error2"}}
+        """#.data(using: .utf8)!
+
+        let error3 = try decoder.decode(JiraService.ServiceError.self, from: errorResponse3)
+        XCTAssertEqual(error3.reason, "msg1 ; msg2 ; key1: error1 ; key2: error2")
```

Comment:
Is that parsed further somehow? Or it's just for display? Because spacing with `" ; "` is a bit weird tbh :)

Reply:
`reason` comes from `protocol Debuggable` conformance, which is a protocol provided by Vapor (well, by SwiftNIO's Core to be precise). It's what's used to display the error in debugger, but also in error responses... and ultimately in Slack's message

---

When we do a request, we call [FailableService.request()](https://github.com/Babylonpartners/Stevenson/blob/master/Sources/Stevenson/Errors.swift#L38) which, as you can see, tries to decode the valid response, but if it fails (`catchFlatMap`), tries instead to parse the response as an error (using the Service's `associatedtype ServiceError`), then throw those (since they conform to `Error`)

Later those errors are captured and wrapped in a `ThrowError` where the `reason` property of those `ServiceError: Debuggable` [are used here](https://github.com/Babylonpartners/Stevenson/blob/master/Sources/Stevenson/Errors.swift#L108-L110) then [used for their `errorDescription`](https://github.com/Babylonpartners/Stevenson/blob/master/Sources/Stevenson/Errors.swift#L147-L149). That's why it ends up also being the text displayed when displaying the errors, including in the Slack response or in the logs.

---

I'm ok to chose a different separator, but I didn't want to use `\n` as a separator in the risk of (1) making the errors too verbose (it was already quite scary to see that many errors when you look at the slack message I linked in the PR description) (2) using a newline would not make it clear that the error messages belong to a single error / point of failure when we then list multiple `ServiceErrors` (like it's the case in the Slack Response where we concatenate errors from an array of `ServiceErrors`, already using `\n` to separate each `ServiceError.reason`, so if `reason` itself contained `\n` it might be too confusing.

I could replace with `" | "` or something similar if it helps? Though to be honest so far in my real tests I only ever saw one item: those error responses seem to most of the time only have one string in `errorMessages` and nothing in `errors`, or only one key/value pair in `errors` and nothing in `errorMessages`... 🤷‍♂

Reply:
I think it's fine for now! When/if there is a real use case we can just reconsider.

There was some kind of practice for gluing separate things like that together (used in something popular at that) but I can't remember how it worked or where it was ¯\_(ツ)_/¯

Reply:
Well one day there will be [ListFormatter](https://developer.apple.com/documentation/foundation/listformatter/3130990-localizedstring) in next version of the stdlib, but not sure when it will become available in Linux, and probably not a good fit for our use case, since that one will likely use commas between all items (and "and" for the final item) in the english locale... so would not be worth it in our use case anyway 😉

Reply:
@karabatov what about this instead?

```
"[1] msg1. [2] msg2. [3] key1: error1. [4] key2: error2."
```
(we would only add those `[n]` if `allErrors.count > 1`)

Reply:
I think that's more readable, yes :)